### DISCUSSION TITLE
USPS Ground Advantage & FedEx Ground Economy Rates

### DISCUSSION THREAD

**Discussion by johnvoncolln** [OpenMage/magento-lts#3300] [General]
Wondering if there are any plans in place to update USPS and FedEx rates to utilize these new existing and upcoming rates?

FedEx: No longer SmartPost, they've changed this to Ground Economy since they are no longer relying on USPS for last mile delivery.
https://www.fedex.com/en-us/shipping/ground/economy.html

USPS: They are combining First-Class Package with Parcel Select Ground to create a new service called Ground Advantage. As far as I know, they are discontinuing both of these separate services, but not sure how the api will react.
https://www.usps.com/ship/ground-advantage.htm

**Comment by fballiano** [comment]
none of my customers use these modules so I can't help here, the only way I can suggest is develop this feature and then create a PR to add it to the core

**Comment by michaelj000** [comment]
It's a simple fix to add USPS Ground Advantage - edit Usa/Model/Shipping/Carrier/Usps.php , in function getCode under 'method' array add:

'1058'   => Mage::helper('usa')->__('USPS Ground Advantage'),

and under service_to_code array add: 

'1058'   => 'Ground Advantage',

Flush cache and the option should appear in the USPS shipping method list in config, select it and it'll work front and admin. 

That worked for me.

**Reply by fballiano** [reply]
I've created https://github.com/OpenMage/magento-lts/pull/3413 not to forget your comment. Should we remove First-Class Package and Parcel Select Ground (actually I don't see parcel select ground in the module)?

**Reply by michaelj000** [reply]
USPS merged First Class Package and Parcel Select Ground into USPS Ground Advantage - so those could be removed (if they are in the module), I noticed the same thing, Parcel Select Ground must have been a later addition (I never used it). So just replacing USPS First Class Package should work.

**Reply by fballiano** [reply]
there are a lot of "first class package":
<img width="537" alt="Screenshot 2023-07-27 alle 18 59 52" src="https://github.com/OpenMage/magento-lts/assets/909743/e0f6887d-7811-4408-9a49-a763318ad552">

should all of them go?

**Reply by michaelj000** [reply]
No - all of the international methods and the other First Class options still apply, it is just the two - First Class Package and Parcel Select Ground - that have changed.

**Reply by fballiano** [reply]
so the first 3, 0_FCP, 53, 61?

**Reply by michaelj000** [reply]
Just 0_FCP and 61, the hold for pickup is still an option with USPS.

**Reply by fballiano** [reply]
done in https://github.com/OpenMage/magento-lts/pull/3413 thank you! it would be great if you could give it a test :-)
